src/Apps/Order/Routes/Payment/index.tsx

```diff
@@ -184,22 +227,50 @@ export const PaymentRoute: FC<PaymentRouteProps> = props => {
     }
   }
 
-  // pushes to the review step, when payment is set with an existing bank account
-  const onBankAccountContinue = () => {
-    props.router.push(`/orders/${props.order.internalID}/review`)
-  }
-
   // fired when balance check is done: either sets error state or moves to /review
-  const onBalanceCheckComplete = (displayInsufficientFundsError: boolean) => {
+  const onBalanceCheckComplete = (
+    displayInsufficientFundsError: boolean,
+    checkResult: BalanceCheckResult
+  ) => {
+    const event = {
+      subject: "checked_account_balance",
+      outcome: checkResult,
+      payment_method: selectedPaymentMethod,
+      currency: order.currencyCode,
+      amount: order.buyerTotalCents,
+      order_id: order.internalID,
+      context_page_owner_type: OwnerType.ordersPayment,
+      flow: order.mode!,
+    }
+
+    trackEvent(event)
+
     setBalanceCheckComplete(true)
     setIsSavingPayment(false)
+
     if (displayInsufficientFundsError) {
       setBankAccountHasInsufficientFunds(true)
       return
     }
+
+    onPaymentStepComplete()
+  }
+
+  const onPaymentStepComplete = () => {
```

Comment:
`onPaymentStepComplete` -> `handlePaymentStepComplete`

Not trying to be pedantic with these naming things! Since React is a functional programming paradigm one of the most crucial things one can do is learn to describe the direction of data flow within the system (this goes for all functional langs). 

- Is the data flow going down? `onFoo={handleFoo}` 
- Is the data being invoked from above? `onFoo()` 
- Is data being invoked locally? `handleFoo()` 
- Is data being set locally? `setFoo()` 
- Is data being set externally from above? `setFoo()` 
- Are we passing a setter down so that it's being invoked below the parent? `onSetFoo={setFoo}` which is then in the child component calling back up to the parent? `onSetFoo(true)`

See how just by naming things consistently one can "see" at a glance exactly which way the data is flowing through a system? These rules only grow in importance the less centralized the state management is.